src/t_hash.c

```diff
@@ -2178,25 +2178,29 @@ static int parseExpireTime(client *c, robj *o, int unit, long long basetime,
 #define HFE_FXX      (1<<6) /* Set fields if all the fields already exist */
 #define HFE_FNX      (1<<7) /* Set fields if none of the fields exist */
 
-/* Parse hsetex command arguments.
- * HSETEX <key>
- *  [FNX|FXX]
- *  [EX seconds|PX milliseconds|EXAT unix-time-seconds|PXAT unix-time-milliseconds|KEEPTTL]
- *  FIELDS <numfields> field value [field value ...]
-*/
-static int hsetexParseArgs(client *c, int *flags,
-                           long long *expire_time, int *expire_time_pos,
-                           int *first_field_pos, int *field_count) {
+/* Command types for unified hash argument parser */
+#define HASH_CMD_HGETEX 0
+#define HASH_CMD_HSETEX 1
+
+/* Parse hash field expiration command arguments for both HGETEX and HSETEX.
+ * HGETEX <key> [EX seconds|PX milliseconds|EXAT unix-time-seconds|PXAT unix-time-milliseconds|PERSIST]
+ *              FIELDS <numfields> field [field ...]
+ * HSETEX <key> [EX seconds|PX milliseconds|EXAT unix-time-seconds|PXAT unix-time-milliseconds|KEEPTTL]
+ *              [FXX|FNX] FIELDS <numfields> field value [field value ...]
+ */
+static int parseHashFieldExpireArgs(client *c, int *flags,
+                                    long long *expire_time, int *expire_time_pos,
+                                    int *first_field_pos, int *field_count,
+                                    int command_type) {
     *flags = 0;
     *first_field_pos = -1;
     *field_count = -1;
     *expire_time_pos = -1;
-
+    int skip_fields = (command_type == HASH_CMD_HSETEX) ? 2 : 1;
     for (int i = 2; i < c->argc; i++) {
         if (!strcasecmp(c->argv[i]->ptr, "fields")) {
             long val;
-
-            if (i >= c->argc - 3) {
+            if (i >= c->argc - 2) {
```

Comment:
why change it? numfields(1) + at least 1 filed and value (2) should be 3?
feel like we should distinguish HGETEX and HSETEX.